synapse/config/workers.py

```diff
@@ -302,46 +301,36 @@ def read_config(self, config, **kwargs):
         # No effort is made to ensure only a single instance of these tasks is
         # running.
         background_tasks_instance = config.get("run_background_tasks_on") or "master"
-        self.run_background_tasks = (
-            self.worker_name is None and background_tasks_instance == "master"
-        ) or self.worker_name == background_tasks_instance
+        self.run_background_tasks = self.instance_name == background_tasks_instance
```

Comment:
suggest explicit parens here for clarity
```suggestion
        self.run_background_tasks = (self.instance_name == background_tasks_instance)
```

Reply:
Black clears those, iirc

Reply:
Yeah it does, not much i can do here